.circleci/config.yml

```diff
@@ -297,41 +321,58 @@ jobs:
             ./nginx_integration_test.sh
 
 
+_workflow_filters:
+  _version_tag: &version_tag
+    # Allows semver or "test" tags, with any suffix.
+    # eg v1.2.3 v0.1.1 v1.2.3-ayy-lmao test test-ayy-lmao
+    # "test" is useful if you want to ensure all workflows run without creating a version.
+    only: /(v[0-9]+\.[0-9]+\.[0-9]|test).*+/
+
+  _run_always: &run_always
+    filters:
+      tags:
+        <<: *version_tag
+
+  _run_on_release_tag: &run_on_release_tag
+    filters:
+      tags:
+        <<: *version_tag
+      branches:
+        ignore: /.*/
+
 workflows:
   version: 2
   build_test_deploy:
     jobs:
-      - build_ubuntu_18_04
-      - build_ubuntu_16_04
-      - test_tsan
-      - test_asan
+      - build_ubuntu_18_04:
+          <<: *run_always
+      - build_ubuntu_16_04:
+          <<: *run_on_release_tag
+      - test_tsan:
+          <<: *run_always
+      - test_asan:
+          <<: *run_always
+      - plugin:
+          requires:
+            - build_ubuntu_18_04
+          <<: *run_on_release_tag
       - package_nginx_module_ubuntu_18_04:
           requires:
             - build_ubuntu_18_04
-          filters:
-            tags:
-              only: /.*/
+          <<: *run_always
       - package_nginx_module_ubuntu_16_04:
           requires:
             - build_ubuntu_16_04
-          filters:
-            tags:
-              only: /.*/
+          <<: *run_on_release_tag
       - package_ubuntu_18_04:
           requires:
             - build_ubuntu_18_04
-          filters:
-            tags:
-              only: /.*/
+          <<: *run_on_release_tag
       - package_ubuntu_16_04:
           requires:
             - build_ubuntu_16_04
-          filters:
-            tags:
-              only: /.*/
+          <<: *run_always
       - integration_test_nginx_ubuntu_18_04:
           requires:
             - package_nginx_module_ubuntu_18_04
-          filters:
-            tags:
-              only: /.*/
+          <<: *run_always
```

Comment:
This is now officially more complicated than the code it builds.

Reply:
What I wanted to do was have two workflows, one called "build_test", the other "build_test_release". First one runs always, second one runs on version tags (or test tags, so we can test it without making a release). 

But circleci doesn't allow per-workflow triggers that aren't cron jobs. So I had to do this instead :'(